phpstan.neon

```diff
@@ -30,6 +30,8 @@ parameters:
 
     ignoreErrors:
         - '#Class "Rector\\Nette\\Rector\\Class_\\MoveFinalGetUserToCheckRequirementsClassMethodRector" is missing @see annotation with test case class reference#'
+        # We need to think how to test rectors like this one
+        - '#Class "Rector\\Nette\\Rector\\Class_\\LatteVarTypesBasedOnPresenterTemplateParametersRector" is missing @see annotation with test case class reference#'
```

Comment:
Test is needed

Reply:
Sure thing :) my question is "how to test rectors which change different file(s)? Maybe they create some file and change the original file too.". Is there any engine for that in rector(-src)?

Reply:
I see. There is a test case that checks the dumped content, but not changed external file.

I would simply compare external content before and after + make a backup of original file to restore it in the end. Filesystem + read/copy/write combo

Reply:
I've added some test, check it please :)